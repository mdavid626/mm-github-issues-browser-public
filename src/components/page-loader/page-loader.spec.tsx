import { cleanup, render, screen } from '@testing-library/react';
import React from 'react';
import PageLoader from './page-loader';

describe('page-loader', () => {
  afterEach(cleanup);

  it('should render', () => {
    const { asFragment } = render(
      <PageLoader>{() => <div>content</div>}</PageLoader>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render spinner', () => {
    const { asFragment } = render(
      <PageLoader isLoading>{() => <div>content</div>}</PageLoader>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render error', () => {
    const { asFragment } = render(
      <PageLoader errorMessage="my error message">
        {() => <div>content</div>}
      </PageLoader>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render error when both loading and error', () => {
    render(
      <PageLoader errorMessage="my error message" isLoading>
        {() => <div>content</div>}
      </PageLoader>
    );
    expect(screen.getByText('my error message')).toBeVisible();
    expect(screen.queryByTestId('PageLoader-spinner')).toBe(null);
  });
});
