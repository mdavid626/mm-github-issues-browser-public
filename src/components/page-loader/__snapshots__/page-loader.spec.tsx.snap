// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page-loader should render 1`] = `
<DocumentFragment>
  <div>
    content
  </div>
</DocumentFragment>
`;

exports[`page-loader should render error 1`] = `
<DocumentFragment>
  <div
    class="PageLoader-errorMessage"
  >
    my error message
  </div>
</DocumentFragment>
`;

exports[`page-loader should render spinner 1`] = `
<DocumentFragment>
  <img
    alt="loading spinner"
    class="PageLoader-spinner"
    data-testid="PageLoader-spinner"
    src="spinner.svg"
  />
</DocumentFragment>
`;
